querybook/webapp/components/AIAssistant/AICommandBar.tsx

```diff
@@ -86,8 +84,23 @@ export const AICommandBar: React.FC<IQueryCellCommandBarProps> = forwardRef(
         } = useCommand(command, commandRunner);
 
         useEffect(() => {
-            setTables((tables) => uniq([...tablesInQuery, ...tables]));
-        }, [tablesInQuery]);
+            setTables((tables) => {
+                if (mentionedTables.length > 0) {
+                    return mentionedTables;
+                } else {
+                    return uniq([...tablesInQuery, ...tables]);
+                }
+            });
+        }, [tablesInQuery, mentionedTables]);
```

Comment:
too many mutation dependencies, how about we start from a fresh slate with a useMemo?